test/unit/modules/blocks/utils.js

```diff
@@ -0,0 +1,260 @@
+			
+			it('should assign logger');
+			it('should assign db');
+			it('should assign dbSequence');
+			it('should assign genesisblock');
+			describe('should assign logic',function () {
+				
+				it('should assign block');
+				
+				it('should assign transaction');
+			});
+		});
+		
+		it('should set self to this');
+		
+		it('should call library.logger.trace("Blocks->Utils: Submodule initialized.")');
+		
+		it('should return self');
+	});
+	
+	describe('readDbRows', function () {
+		
+		it('should loop through all given rows');
+		
+		it('should call library.logic.block.dbRead with current row');
+
+		describe('when block exists', function () {
+
+			describe('and block is not already in list', function () {
+				
+				describe('and block is the genesis block', function () {
+					
+					it('should generate a fake signature for it');		
+				});
+
+				it('should push the block.id to order array');
+
+				it('should add the block to block array');
+			});
+			
+			it('should call library.logic.transaction.dbRead with current row');
+			
+			describe('and there are no transactions in the block', function () {
+				
+				it('should set transactions to an empty object');		
+			});
+			
+			describe('and a transaction exists', function () {
+				
+				it('should be added to the block if it is not added yet');				
+			});
+		});
+		
+		it('should reorganize the block array');
+		
+		it('should return the block array');		
+	});
+	
+	describe('loadBlocksPart', function () {
+		
+		describe('when there is no error when loading block array', function () {
+			
+			it('should be normalized');			
+		});
+		
+		it('should return callback with the error object and blocks as params');
+	});
+	
+	describe('loadLastBlock', function () {
+		
+		it('should call library.dbSequence.add with callback');
+		
+		it('should call library.db.query to load the last block');
+		
+		describe('when db query fails', function () {
+			
+			it('should call logger.error with error stack');
+
+			it('should call callback with the Blocks#loadLastBlock error');
+		});
+
+		describe('when db.query succeeds', function () {
+		
+			it('should call modules.blocks.utils.readDbRows and set "block" to first item of return value');
+
+			describe('sorting the block-transactions array', function () {
+
+				describe('when block.id equals genesis.block.id', function () {
+					
+					describe('and transactionType equals VOTE', function () {
+
+						it('should move it to the end of the block array');
+					});
+				});
+
+				describe('when transactionType equals SIGNATURE', function () {
+
+					it('should move it to the end of the block.transactions array');
+				});
+			});
+
+			it('should call modules.blocks.lastBlock.set(block)');
+
+			it('should return callback with error = null');
+
+			it('should return callback with result containing the block');			
+		});
+	});
+	
+	describe('getIdSequence', function () {
+		
+		it('should set lastBlock to modules.blocks.lastBlock.get()');
```

Comment:
private variable - impossible to check
add a test for check if modules.blocks.lastBlock.get was called instead